recipes/approvaltests.cpp/all/conanfile.py

```diff
@@ -66,3 +68,17 @@ def package_id(self):
 
     def _boost_test_supported(self):
         return Version(self.version) >= "8.6.0"
+
+    def _std_puttime_required(self):
+        return Version(self.version) >= "10.2.0"
+
+    def _validate_compiler_settings(self):
+        compiler = str(self.settings.compiler)
+        version = tools.Version(self.settings.compiler.version)
+
+        if self._std_puttime_required():
+            self._check_compiler_version_not_less_than(compiler, version, "gcc", "5")
+
+    def _check_compiler_version_not_less_than(self, my_compiler, my_version, compiler, compiler_min_version):
+        if my_compiler == compiler and my_version < compiler_min_version:
+            raise ConanInvalidConfiguration(f"Version {self.version} of ApprovalTests.cpp does not support {compiler} < {compiler_min_version}")
```

Comment:
Hey, I think my suggested code below is more readable because:
- I think having a function named `not_less_then` is not fast to understand, it's similar to writing `not <`.
  I believe `require at least` is more readable: `>=`.
- Stringification of `compiler` is not needed, and unwanted.
  You can compare against `SettingsItem` objects.
  Also, by not stringizing the object, conan can check whether the compiler name is a valid one (this catches errors).
- We at cci live in the past decade and don't support f-strings yet :smile: 

```suggestion
        if self._std_puttime_required():
            self._require_at_least_compiler_version("gcc", 5)

    def _require_at_least_compiler_version(self, compiler, compiler_version):
        if self.settings.compiler == compiler and tools.Version(self.settings.compiler.version) < compiler_version:
            raise ConanInvalidConfiguration("{}/{} with compiler {} requires at least compiler version {}".format(self.name, self.version, compiler, compiler_version))
```